Issue #18 - ThemesFold in Settings

giro39 opened the pull request.

giro39 committed:
Add basic styling

giro39 committed:
Add basic styling

giro39 committed:
Add flex wrap, add first breakpoint styling

giro39 committed:
Add flex wrap, add first breakpoint styling

giro39 committed:
Add ThemeContext

giro39 committed:
Add ThemeContext

Aeri4a committed:
rebase main & layout fix

giro39 committed:
Merge branch '18-settings-themes' of https://github.com/akai-org/akaitodo-frontend into 18-settings-themes

giro39 committed:
Add themes functionality

giro39 committed:
Add mode slider

As the themes started to work, a lot of things came out to be improved - the values of the Sass variables will have to be reworked.

giro39 committed:
Add mobile view

giro39 committed:
Add new vars, change few theme ideas

giro39 committed:
Update settings to use vars, update red theme

giro39 committed:
Add localStorage handling

giro39 committed:
Fix slider reset bug, edit some vars

giro39 committed:
Fix padding, fix third color overflow

giro39 committed:
Minor corrections in styling

Aeri4a commented on src/components/settings/ThemesFold.jsx:
```diff
@@ -1,9 +1,162 @@
+        [setTheme],
+    );
+
+    const colorPalettes = colors.map(
```
Nice mapping

Aeri4a commented on src/styles/_vars.scss:
```diff
@@ -4,31 +4,280 @@
-    // todo
+    // -= RED COLOR =-
+    [themeStyle='red'] {
+        --darkgray-color: hsl(0, 0%, 64%);
+        --lightgray-color: hsl(0, 0%, 85%);
```
If this is repeating every x-lines, then I think we could move it to the top under the root.

Aeri4a commented on src/styles/components/settings/ThemesFold.module.scss:
```diff
@@ -0,0 +1,203 @@
+@use '../../vars' as *;
+
```
I checked on live view and I found two things with this style:
1. main container is not getting 100% height
![Screenshot from 2024-09-14 22-20-21](https://github.com/user-attachments/assets/2dd5112a-1b2b-4fcb-97a3-92d42ddd13b3)

2. Mode button is on top of mobile menu in open state
![image](https://github.com/user-attachments/assets/de1f9ff6-8e36-444e-90ce-d59eaa41d3cd)

Aeri4a commented on src/styles/layout/ScrollableContainer/ScrollableContainer.module.scss:
```diff
@@ -10,7 +10,8 @@
     flex: 1;
     flex-direction: column;
-    padding: 10px;
+    // padding: 10px;
```
Remove unused

Aeri4a reviewed (changes requested):
Overall - Good Job! :D
Just some small things I would say.

giro39 committed:
Create utils folders, outsource code to utils

giro39 committed:
Delete repeating vars, update styling to cover full page

giro39 committed:
Fix getStoredTheme bug, change the mode button

- the bug was that the function in utils did not have access to the initial value of the ThemeContext
- Replaced the checkbox toggle for theme mode (LIGHT/DARK) with a flip animation effect

Aeri4a reviewed (approved):
Looks good for me. Good job! :)) You can do squash & merge and then please pull whole main and test it again.
